### PR TITLE
Enable bala tests for function pointers

### LINKED ISSUE #29112: Enable bala tests for function pointers
**Description:**
$title.
This issue created to enable bala tests for function pointer(https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bala/functions/FunctionPointersTest.java) tests and add negative bala tests for `function` type descriptor.

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #29112 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import testorg/foo;
+import testorg/foo as foo;
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal:1]: We don't need this, right?
```suggestion
import testorg/foo;
```

File: tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers_negative.bal [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import testorg/foo as foo;
+
+function functionWithFunctionReturnType() returns function {
+    return foo:getGlf1();
+}
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers_negative.bal:21]: Nothing is tested using this?

File: tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal (hunk 1 above)
Comment (chiranSachintha) [reply tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal:1]: Yes. But it's clear(for me :smiley:).

Comment (MaryamZi) [reply tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal:1]: Oh okay, :smiley:  But if you think of how we get the prefix (part of the module name after the last period) and given that  the module name has no periods here, there's no difference, right?

This is the same as saying
```suggestion
import ballerina/io as io;
```

File: tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers_negative.bal (hunk 2 above)
Comment (chiranSachintha) [reply tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers_negative.bal]: Fixed.

File: tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal (hunk 1 above)
Comment (chiranSachintha) [reply tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_global_function_pointers.bal]: Removed.:relaxed:

**Review by hasithaa — APPROVED** [review]

**Pull request merged**